torch/ao/pruning/_experimental/pruner/README.md

```diff
@@ -1,93 +1,263 @@
-# Intro
+# Structured Pruning
+
+## Intro / Motivation
+
+**pruning** is the process of removing weights / connections in a neural network in order to make it run faster.
+
+There are two types of pruning that we support: **structured pruning** and **unstructured pruing**
+
+* **Unstructured pruning** refers to pruning any arbritary weight
+* **Structured pruning** refers to pruning by removing entire rows / columns of a weight.
+
+Structured pruning offers several advantages over unstructured pruning at the cost of lower granuality:
+1. We can store the masks more efficeintly in structured pruning.
+2. Structured pruning relies on removing columns / rows to a speedup whereas unstructured pruning requires custom kernels.
+
+Pruning is usually followed by a retraining or fine-tuning step.
+
+## What do I need to use Structured Pruning?
+
+**Your model must be FX symbolically traceable**
+
+We support structured pruning for the following patterns out of the box.
+- linear -> linear
+- linear -> activation -> linear
+- conv2d -> conv2d
+- conv2d -> activation -> conv2d
+- conv2d -> activation -> pool -> conv2d
+- conv2d -> pool -> activation -> conv2d
+- conv2d -> adaptive pool -> flatten -> linear
+
+If you are looking to prune a different pattern, for example `(conv2d -> activation -> pool -> dropout -> conv2d)`, you can do this by creating modifying the pattern dict that we provide to the pruner, see [here](#writing-custom-patterns-and-pruning-functions-for-structured-pruning)
+
+It is up to the user to define the following two items.
+1. What layers are you looking to prune
```

Comment:
seems like this may be better covered in the link above about custom pruning, feels a bit out of place, especially without any more details